### ISSUE TITLE
About Batch Size

### ISSUE THREAD

**Issue by botaoye** [jd730/OICR-pytorch#18]
Hi, I read the code roughly and found something strange about the training batch size. you set batch_size=4, but in line 26 of lib/oicr_layer/layer.py, you assert batch size equal to 1.
                              `assert num_images == 1, 'batch size shoud be equal to 1`
So, why you can set batch_size=4 with this assert?

Besides, have tried Resnet backbone? I use Res50 as backbone, but it doesn't work well, the VOC2007 AP only get 0.27.
Thanks very much!

**Comment by jd730** [comment]
Hi @suyeH, 
My code assumes 1 image comes in a GPU. This is because we are using about 2,000 object proposals from Selective Search, which requires ~10GB.

However, the batch size of the original implementation (written in Caffe) is 2. Here is a detail,
it runs backward for every two steps but it does not divide the loss into two.
While reproducing it, I found that I should follow this strategy. ( It should be same as learning rate 0.002 with two GPUs but I've never tried.)
This is the reason why I set https://github.com/jd730/OICR-pytorch/blob/97c8ecccfb186f35acd89f4537e5f45a245ed817/trainval_net.py#L252

If you are using Titan RTX, Quadro 8000, and other GPUs having large storage, you might change `lib/oicr_layer/layer.py`.

I remember that I tried to use ResNet50 backbone and I found that bath size is really critical. Usually, batch normalization requires at least 8 batches and has worse performance on smaller batch sizes. That is why you get worse performance than VGG16.
You can use other normalization techniques such as Group Normalization, Instance Normalization.

**Comment by botaoye** [comment]
Ok, I get it. Thanks for your answer and suggestion.
